sonic-xcvrd/scripts/xcvrd

```diff
@@ -0,0 +1,452 @@
+        
+        logical_port_list = platform_sfputil.logical
+        for logical_port_name in logical_port_list:
+            post_port_dom_info_to_db(logical_port_name, self.dom_table)
+        
+        self.task_timer = threading.Timer(DOM_INFO_UPDATE_PERIOD_SECS, self.task_run)
+        self.task_timer.start()
+
+    def task_stop(self):
+        self.task_running_event.clear()
+        self.task_timer.join()
+
+#=============================== Main ================================
+
+def main():
+    log_info("Starting up...")
+
+    # Register our signal handlers
+    signal.signal(signal.SIGHUP, signal_handler)
+    signal.signal(signal.SIGINT, signal_handler)
+    signal.signal(signal.SIGTERM, signal_handler)
+
+    # Load platform-specific sfputil class
+    err = load_platform_sfputil()
+    if err != 0:
+        log_error("failed to load sfputil")
+        sys.exit(1)
+
+    # Load port info
+    try:
+        port_config_file_path = get_path_to_port_config_file()
+        platform_sfputil.read_porttab_mappings(port_config_file_path)
+        # platform_sfputil.read_port_mappings()
+    except Exception, e:
+        log_error("Error reading port info (%s)" % str(e), True)
+        sys.exit(2)
+
+    # Connect to STATE_DB and create transceiver info/dom info table
+    state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_MSECS)
+    int_tbl = swsscommon.Table(state_db, "TRANSCEIVER_INFO")
+    dom_tbl = swsscommon.Table(state_db, "TRANSCEIVER_DOM_SENSOR")
+
+    # Connect to APPL_DB abd subscribe to PORT table notifications
+    appl_db = swsscommon.DBConnector(swsscommon.APPL_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_MSECS)
+
+    sel = swsscommon.Select()
+    sst = swsscommon.SubscriberStateTable(appl_db, swsscommon.APP_PORT_TABLE_NAME)
+    sel.addSelectable(sst)
+
+    # Make sure this daemon started after all port configured.
+    while True:
+        (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
+        if state == swsscommon.Select.TIMEOUT:
+            continue
+        if state != swsscommon.Select.OBJECT:
+            log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+            continue
+
+        (key, op, fvp) = sst.pop()
+        if key in ["PortConfigDone", "PortInitDone"]:
+            break
+
+    # Post all the current interface SFP info to STATE_DB
+    logical_port_list = platform_sfputil.logical
+    for logical_port_name in logical_port_list:
+        post_port_sfp_info_to_db(logical_port_name, int_tbl)
+        post_port_dom_info_to_db(logical_port_name, dom_tbl)
+
+    # Start the dom sensor info update timer thread
+    dom_update_threade_event = threading.Event()
+    dom_update_thread_timer = None
```

Comment:
event and timer objects could be hidden in the update class.

Reply:
revised.